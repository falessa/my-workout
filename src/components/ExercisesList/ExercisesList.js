import React from 'react';
import { StyleSheet, View, FlatList } from 'react-native';
import ExerciseCard from '../ExerciseCard/ExerciseCard';
import PlanCardsSeparator from '../PlanCardsSeparator'

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'stretch',
        justifyContent: 'center',
        width: '85%',
    }
})

const ExercisesList = ({ exercises }) => {
    return (
        <View style={styles.container}>
            <FlatList
                data={exercises}
                keyExtractor={x => String(x.name)}
                renderItem={({ item }) => <ExerciseCard exerciseData={item}/> }
                ItemSeparatorComponent={PlanCardsSeparator}
            />
        </View>
    )
}

export default ExercisesList;