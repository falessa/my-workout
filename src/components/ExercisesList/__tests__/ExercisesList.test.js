import "react-native";
import React from "react";
import { create } from "react-test-renderer";
import ExercisesList from '../ExercisesList';

describe('ExercisesList tests', () => {
    const props = {
        exercises: [
            {
                name: "Squats",
                sets: "4",
                repetitions: "12",
                kgs: "5"
            },
            {
                name: "Pushups",
                sets: "3",
                repetitions: "15",
                kgs: "2"
            }
        ]
    }

    const wrapper = create(<ExercisesList {...props} />)

    test('Render ExercisesList', () => {
        expect(wrapper).toMatchSnapshot()
    })
})