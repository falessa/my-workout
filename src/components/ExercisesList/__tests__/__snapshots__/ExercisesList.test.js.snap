// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExercisesList tests Render ExercisesList 1`] = `
<View
  style={
    {
      "alignItems": "stretch",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "width": "85%",
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    data={
      [
        {
          "kgs": "5",
          "name": "Squats",
          "repetitions": "12",
          "sets": "4",
        },
        {
          "kgs": "2",
          "name": "Pushups",
          "repetitions": "15",
          "sets": "3",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#eee",
              "borderRadius": 10,
              "flex": 1,
              "justifyContent": "center",
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 17,
                "fontWeight": "bold",
              }
            }
          >
            Squats
          </Text>
          <View
            style={
              {
                "alignContent": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 5,
              }
            }
          >
            <View
              getItem={[Function]}
              getItemCount={[Function]}
              keyExtractor={[Function]}
              onContentSizeChange={[Function]}
              onLayout={[Function]}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={50}
              stickyHeaderIndices={[]}
              viewabilityConfigCallbackPairs={[]}
            />
          </View>
        </View>
        <View
          style={
            {
              "height": 15,
            }
          }
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#eee",
              "borderRadius": 10,
              "flex": 1,
              "justifyContent": "center",
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 17,
                "fontWeight": "bold",
              }
            }
          >
            Pushups
          </Text>
          <View
            style={
              {
                "alignContent": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 5,
              }
            }
          >
            <View
              getItem={[Function]}
              getItemCount={[Function]}
              keyExtractor={[Function]}
              onContentSizeChange={[Function]}
              onLayout={[Function]}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={50}
              stickyHeaderIndices={[]}
              viewabilityConfigCallbackPairs={[]}
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
